module/lua/ldo.c

```diff
@@ -29,6 +29,18 @@
 
 
 
+/* Return the number of bytes available on the stack. */
+#if defined (_KERNEL) && defined(__linux__)
+#include <asm/current.h>
+static intptr_t stack_remaining(void) {
+  char local;
+  return (intptr_t)(&local - (char *)current->stack);
+}
+#else
```

Comment:
I'll see if I can come up with something that works for FreeBSD.

Reply:
```C
#include <sys/pcpu.h>
struct thread *td = curthread;
return (intptr_t)((char *)&td - (char *)td->td_kstack);
```

Reply:
Thanks!  I've added this to the PR.  I changed the code slightly so that both linux and freebsd have the same form with the `local`, take a look.